doc/indexing.rst

```diff
@@ -191,39 +194,126 @@ index labels along a dimension dropped:
+
+    data = xray.DataArray([1, 2, 3], dims='x')
+    data.sel(x=[1.1, 1.9], method='nearest')
+    data.sel(x=0.1, method='backfill')
+    data.reindex(x=[0.5, 1, 1.5, 2, 2.5], method='pad')
 
-Attributes are persisted in array indexing:
+Using ``method='nearest'`` or a scalar argument with ``.sel()`` requires pandas
+version 0.16 or newer.
+
+The method parameter is not yet supported if any of the arguments
+to ``.sel()`` is a ``slice`` object:
+
+.. ipython::
+    :verbatim:
+
+    In [1]: data.sel(x=slice(1, 3), method='nearest')
+    NotImplementedError
+
+However, you don't need to use ``method`` to do inexact slicing. Slicing
+already returns all values inside the range (inclusive), as long as the index
+labels are monotonic increasing:
+
+.. ipython:: python
+
+    data.sel(x=slice(0.9, 3.1))
+
+Indexing axes with monotonic decreasing labels also works, as long as the
+``slice`` or ``.loc`` arguments are also decreasing:
 
 .. ipython:: python
 
-    arr2 = arr.copy()
-    arr2.attrs['units'] = 'meters'
-    arr2[0, 0].attrs
+    reversed_data = data[::-1]
+    reversed_data.loc[3.1:0.9]
+
+.. _masking with where:
+
+Masking with ``where``
+----------------------
+
+Indexing methods on xray objects generally return a subset of the original data.
+However, it is sometimes useful to select an object with the same shape as the
+original data, but with some elements masked. To do this type of selection in
+xray, use :py:meth:`~xray.DataArray.where`:
+
+.. ipython:: python
+
+    arr = xray.DataArray(np.arange(16).reshape(4, 4), dims=['x', 'y'])
+    arr.where(arr.x + arr.y < 4)
+
+This is particularly useful for ragged indexing of multi-dimensional data,
+e.g., to apply a 2D mask to an image. Note that ``where`` follows all the
+usual xray broadcasting and alignment rules for binary operations (e.g.,
+``+``) between the object being indexed and the condition, as described in
+:ref:`comput`:
+
+.. ipython:: python
+
+    arr.where(arr.y < 2)
+
+Multi-dimensional indexing
+--------------------------
+
+Xray does not yet support efficient routines for generalized multi-dimensional
+indexing or regridding. However, we are definitely interested in adding support
+for this in the future (see :issue:`475` for the ongoing discussion).
+
+.. _copies vs views:
+
+Copies vs. views
+----------------
+
+Whether array indexing returns a view or a copy of the underlying
+data depends on the nature of the labels. For positional (integer)
+indexing, xray follows the same rules as NumPy:
+
+* Positional indexing with only integers and slices returns a view.
+* Positional indexing with arrays or lists returns a copy.
+
+The rules for label based indexing are more complex:
+
+* Label-based indexing with only slices returns a view.
+* Label-based indexing with arrays returns a copy.
+* Label-based indexing with scalars returns a view or a copy, depending
+  upon if the corresponding positional indexer can be represented as an
+  integer or a slice object.
```

Comment:
unfortunately, the distinction here is pretty complex and I'm not even sure I understand it fully...